src/result_processor.c

```diff
@@ -59,13 +59,18 @@ static int RPGeneric_NextEOF(ResultProcessor *rp, SearchResult *res) {
 typedef struct {
   ResultProcessor base;
   IndexIterator *iiter;
+  struct timespec timeout;
 } RPIndexIterator;
 
 /* Next implementation */
 static int rpidxNext(ResultProcessor *base, SearchResult *res) {
   RPIndexIterator *self = (RPIndexIterator *)base;
   IndexIterator *it = self->iiter;
 
+  if (TimedOut(self->timeout) == RS_RESULT_TIMEDOUT) {
```

Comment:
@ashtul if you put the timedout only on the results processor, things like group by or sort that need to accumulate all the results might cause as miss the timeout, I suggest to create a macro that checks it each call to any results processor (look how it was implemented on 1.4)

Reply:
@MeirShpilraien , As discussed, the timeout will be on collecting the results only and not later processing. It is documented.

The reasons are:
* The complexity while querying the index can be high O(n^2) and create delays while other result processors work in O(1) so even if the query exceeds the timeout, it would not be significant.
* We might reach timeout after returning the number of results found. Changing this behavior will be a breaking change.